python/social_bot/teacher.py

```diff
@@ -79,6 +81,16 @@ def run(self, agent, world):
         """
         pass
 
+    @abstractmethod
+    def task_specific_observation(self):
+        """
+        Args:
+            None
+        Returns:
+            np.array of the extra observations should be added into the
+            observation besides self states, for the non-image case
+        """
+        pass
```

Comment:
Since  subclass does have to implement this, we can return [] here and remove `@abstractmethod` decorator and comment that subclass can override this